Bundle/WorkflowBundle/DependencyInjection/WorkflowExtension.php

```diff
@@ -105,13 +112,80 @@ private function registerWorkflowConfiguration(array $workflows, ContainerBuilde
                 $workflowDefinition->replaceArgument(1, $markingStoreDefinition);
             }
             $workflowDefinition->replaceArgument(3, $name);
+
             // Store to container
             $workflowId = sprintf('%s.%s', $type, $name);
             $container->setDefinition($workflowId, $workflowDefinition);
             $container->setDefinition(sprintf('%s.definition', $workflowId), $definitionDefinition);
+
             // Add workflow to Registry
             foreach ($workflow['supports'] as $supportedClass) {
-                $registryDefinition->addMethodCall('add', array(new Reference($workflowId), $supportedClass));
+
+            }
+
+            // Add workflow to Registry
+            if ($workflow['supports']) {
+                foreach ($workflow['supports'] as $supportedClassName) {
+                    if (class_exists(Workflow\SupportStrategy\ClassInstanceSupportStrategy::class)) {
+                        $strategyDefinition = new Definition(Workflow\SupportStrategy\ClassInstanceSupportStrategy::class, array($supportedClassName));
+                        $strategyDefinition->setPublic(false);
+                        $registryDefinition->addMethodCall('add', array(new Reference($workflowId), $strategyDefinition));
+                    } else {
+                        $registryDefinition->addMethodCall('add', array(new Reference($workflowId), $supportedClassName));
+                    }
+                }
+            } elseif (isset($workflow['support_strategy'])) {
+                $registryDefinition->addMethodCall('add', array(new Reference($workflowId), new Reference($workflow['support_strategy'])));
+            }
+
+            // Enable the AuditTrail
+            if ($workflow['audit_trail']['enabled']) {
+                $listener = new Definition(Workflow\EventListener\AuditTrailListener::class);
+                $listener->addTag('monolog.logger', array('channel' => 'workflow'));
+                $listener->addTag('kernel.event_listener', array('event' => sprintf('workflow.%s.leave', $name), 'method' => 'onLeave'));
+                $listener->addTag('kernel.event_listener', array('event' => sprintf('workflow.%s.transition', $name), 'method' => 'onTransition'));
+                $listener->addTag('kernel.event_listener', array('event' => sprintf('workflow.%s.enter', $name), 'method' => 'onEnter'));
+                $listener->addArgument(new Reference('logger'));
+                $container->setDefinition(sprintf('%s.listener.audit_trail', $workflowId), $listener);
+            }
+
+            if (!class_exists('Symfony\Component\Workflow\EventListener\GuardListener')) {
+                continue;
+            }
+
+            // Add Workflow Expression Language
+            $exLanguageDefinition = new Definition(Workflow\EventListener\ExpressionLanguage::class);
```

Comment:
Since ExpressionLanguage and GuardListener classes become a part of workflow component only since 3.3.2 and minimum supported version is 3.2.0, i think we should  throw exception when there is an attempt to use guard expressions without these classes defined in order to keep BC